pkg/server/server_sql.go

```diff
@@ -1306,13 +1284,44 @@ func (s *SQLServer) preStart(
 		if err := s.tenantConnect.Start(ctx); err != nil {
 			return err
 		}
-		// Confirm tenant exists prior to initialization. This is a sanity
-		// check for the dev environment to ensure that a tenant has been
-		// successfully created before attempting to initialize a SQL
-		// server for it.
-		if err := checkTenantExists(ctx, s.execCfg.Codec, s.execCfg.DB); err != nil {
+	}
+
+	// Load the multi-region enum by reading the system database's descriptor.
+	// This also serves as a simple check to see if a tenant exist (i.e. by
+	// checking whether the system db has been bootstrapped).
+	var mrEnumTypeDesc catalog.TypeDescriptor
+	if err := s.internalExecutorFactory.DescsTxn(ctx, s.execCfg.DB, func(
+		ctx context.Context, txn *kv.Txn, collection *descs.Collection,
+	) error {
+		mrEnumTypeDesc = nil // reset for retry
+		flags := tree.CommonLookupFlags{
+			Required:    true,
+			AvoidLeased: true,
+		}
+		_, systemDB, err := collection.GetImmutableDatabaseByID(
+			ctx, txn, keys.SystemDatabaseID, flags,
+		)
+		if err != nil {
+			return errors.Wrap(err, "could not retrieve system DB")
+		}
+		if !systemDB.IsMultiRegion() {
+			return nil
+		}
+		id, err := systemDB.MultiRegionEnumID()
+		if err != nil {
 			return err
 		}
+		mrEnumTypeDesc, err = collection.GetImmutableTypeByID(ctx, txn, id, tree.ObjectLookupFlags{
+			CommonLookupFlags: flags,
+		})
+		return err
+	}); err != nil {
+		return err
+	}
```

Comment:
We have a `SynthesizeRegionConfig` function in `sql` which does most of this already, would it be possible to use it? You'd have to extend `multiregion.RegionConfig` to include the physical representation, presumably, but that seems reasonable.